src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,282 @@
+            newY = checkIfPlaceable(x3,y,z3,world);
+            final BlockPos corner4 = new BlockPos(x3, newY, z3);
+            world.setBlockState(corner1, constructionTapeCorner.withProperty(FACING, EnumFacing.SOUTH));
+            world.setBlockState(corner2, constructionTapeCorner.withProperty(FACING, EnumFacing.EAST));
+            world.setBlockState(corner3, constructionTapeCorner.withProperty(FACING, EnumFacing.WEST));
+            world.setBlockState(corner4, constructionTapeCorner.withProperty(FACING, EnumFacing.NORTH));
+        }
+
+    /**
+     ** Proxy to remove the tape also with the building only.
+     * @param building the building.
+     * @param world the world.
+     */
+    public static void removeConstructionTape(@NotNull AbstractBuilding building, @NotNull World world)
+    {
+        removeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+    /**
+     * Remove construction tape.
+     * @param workOrder the workorder.
+     * @param world the world.
+     */
+
+    public static void removeConstructionTape(@NotNull WorkOrderBuild workOrder,@NotNull World world)
+    {
+        final StructureWrapper wrapper = new StructureWrapper(world, workOrder.getStructureName());
+        final BlockPos pos = workOrder.getBuildingLocation();
+        int tempRotation = 0;
+        if (workOrder.getRotation() == 0 && !(workOrder instanceof WorkOrderBuildDecoration))
+        {
+            final IBlockState blockState = world.getBlockState(pos);
+            if (blockState.getBlock() instanceof AbstractBlockHut)
+            {
+                tempRotation = BlockUtils.getRotationFromFacing(blockState.getValue(AbstractBlockHut.FACING));
+            }
+        }
+        else
+        {
+            tempRotation = workOrder.getRotation();
+        }
+        wrapper.rotate(tempRotation, world, workOrder.getBuildingLocation());
+        wrapper.setPosition(pos);
+        final int x1 = wrapper.getPosition().getX() - wrapper.getOffset().getX() - 1;
+        final int z1 = wrapper.getPosition().getZ() - wrapper.getOffset().getZ() - 1;
+        final int x3 = wrapper.getPosition().getX() + (wrapper.getWidth() - wrapper.getOffset().getX());
+        final int z3 = wrapper.getPosition().getZ() + (wrapper.getLength() - wrapper.getOffset().getZ());
+        if (x1 < x3)
+        {
+            for (int i = x1; i <= x3; i++)
+            {
+                final BlockPos block1 = new BlockPos(i, 0, z1);
+                final BlockPos block2 = new BlockPos(i, 0, z3);
+                removeTapeIfNecessary(world,block1,ModBlocks.blockConstructionTape);
+                removeTapeIfNecessary(world,block2,ModBlocks.blockConstructionTape);
+            }
+        }
+        else
+        {
+            for (int i = x3; i <= x1; i++)
+            {
+                final BlockPos block1 = new BlockPos(i, 0, z1);
+                final BlockPos block2 = new BlockPos(i, 0, z3);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00117)